bin/ruby-build

```diff
@@ -1268,8 +1268,13 @@ build_package_enable_yjit() {
     fi
   fi
 
-  # If YJIT is explicitly specified, don't change anything
-  if [[ " ${RUBY_CONFIGURE_OPTS} " != *" --enable-yjit"* ]] && [[ " ${RUBY_CONFIGURE_OPTS} " != *" --disable-yjit"* ]]; then
+  x86_arch=""
+  if [[ "$(uname -m)" == "x86_64" ]]; then
+    x86_arch="1"
+  fi
```

Comment:
Isn't it much simpler to just `return`?
Like `[[ "$(uname -m)" != "x86_64" ]] && return 0` (or with a `if`)

Reply:
Might also want to check + return for `RUBY_CONFIGURE_OPTS`, so we don't have nested `if` at all.

BTW, those checks should be done before we check `rustc`, so we don't take extra time if anyway we're not going to use the result of `rustc --version`

Reply:
Okay. I've restructured this.